blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -1,204 +1,113 @@
-            if (scmConfig.getConfig().get("orgName") instanceof String) {
-                orgName = (String) scmConfig.getConfig().get("orgName");
-            }
-            if (scmConfig.getConfig().get("repos") instanceof List) {
-                for (String r : (List<String>) scmConfig.getConfig().get("repos")) {
-                    repos.add(r);
+    @Nullable
+    @Override
+    protected AbstractScmSourceEvent getScmSourceEvent(@Nonnull final MultiBranchProject project, @Nonnull SCMSource source) {
+        if(source instanceof GitHubSCMSource) {
+            return new AbstractScmSourceEvent(((GitHubSCMSource)source).getRepository(),
+                    ((GitHubSCMSource)source).getApiUri()) {
+                @Override
+                public boolean isMatch(@NonNull SCMSource source) {
+                    SCMSourceOwner sourceOwner = source.getOwner();
+                    return ((GitHubSCMSource)source).getRepository().equals(getSourceName()) && sourceOwner != null
+                            && sourceOwner.getFullName().equals(project.getFullName());
                 }
-            }
+            };
         }
+        return null;
+    }
 
-        validateCredentialId(credentialId, apiUrl);
-        // TODO: validate the endpoint already exists? validateCredentialId is doing that somewhat implicitly above
-        updateEndpoints(apiUrl);
-        String singleRepo = repos.size() == 1 ? repos.get(0) : null;
-
-        ModifiableTopLevelItemGroup orgRoot = getParent(organization);
-        
-        Item item = Jenkins.getInstance().getItemByFullName(orgRoot.getFullName() + '/' + orgName);
-
-        boolean creatingNewItem = item == null;
-        try {
-            if (item == null) {
-                item = createProject(getName(), DESCRIPTOR, CustomOrganizationFolderDescriptor.class, organization);
+    @Override
+    protected List<ErrorMessage.Error> validate(String name, BlueScmConfig scmConfig) {
+        List<ErrorMessage.Error> errors = Lists.newArrayList();
+        StandardUsernamePasswordCredentials credentials = null;
+        String credentialId = computeCredentialIdWithGithubDefault(scmConfig);
+        if(StringUtils.isBlank(scmConfig.getUri())){
+            errors.add(new ErrorMessage.Error("scmConfig.uri",
+                    ErrorMessage.Error.ErrorCodes.MISSING.toString(),
+                    "uri is required"));
+        }
+
+        if(credentialId != null){
+            credentials = CredentialsUtils.findCredential(credentialId, StandardUsernamePasswordCredentials.class, new BlueOceanDomainRequirement());
+        }
+        if (credentials == null) {
+            errors.add(new ErrorMessage.Error("scmConfig.credentialId",
+                    ErrorMessage.Error.ErrorCodes.NOT_FOUND.toString(),
+                    "No Credentials instance found for credentialId: " + credentialId));
+        }else if(StringUtils.isBlank(scmConfig.getUri())){
```

Comment:
This else if statement seems odd since the condition is checked above on line 78 and throws an exception if true?